### PR TITLE
Tests for fileAssociations updates

### LINKED ISSUE #143: Changing the file associations or catalogs in the preferences should trigger validation
When a file association or a catalog is added or modified in the preferences, if a matching document is already open, it doesn't take into account the new schema.
Either diagnostics on opened documents are stale or missing.
So changing associations or catalogs should trigger validation for all opened documents.

### PR DESCRIPTION
**Pull request by NikolasKomonen**
Closes #143

### CODE REVIEW

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/contentmodel/ContentModelPlugin.java [hunk 1]
```diff
@@ -14,6 +14,7 @@
 import org.eclipse.lsp4xml.services.extensions.IXMLExtension;
 import org.eclipse.lsp4xml.services.extensions.XMLExtensionsRegistry;
 import org.eclipse.lsp4xml.services.extensions.save.ISaveContext;
+import org.eclipse.lsp4xml.settings.InitializationOptionsSettings;
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/contentmodel/ContentModelPlugin.java]: new import seems unused

File: org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/XMLAssert.java [hunk 2]
```diff
@@ -269,6 +269,30 @@ public static void testDiagnosticsFor(String xml, String catalogPath, Consumer<X
...
+		XMLProblems problems = new XMLProblems();
+		problems.setNoGrammar("ignore");
+		settings.setProblems(problems);
+		if (catalogPath != null) {
+			// Configure XML catalog for XML schema
+			settings.setCatalogs(new String[] { catalogPath });
+		}
+		xmlLanguageService.doSave(new SettingsSaveContext(settings));
+		
+		//assertDiagnostics(actual, Arrays.asList(expected), filter);
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/XMLAssert.java]: so nothing is being asserted here

**Comment by NikolasKomonen** [comment]
@fbricon updated

**Comment by fbricon** [comment]
so validationOnRootDoesntUpdate checks fileassociation was updated, and validationOnRootUpdate checks it was not. I'm confused. Looks like the test methods are poorly named

**Comment by NikolasKomonen** [comment]
@fbricon updated the test names

File: org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/XMLAssert.java [hunk 3]
```diff
@@ -268,7 +268,7 @@ public static void testDiagnosticsFor(String xml, String catalogPath, Consumer<X
 		assertDiagnostics(actual, Arrays.asList(expected), filter);
 
 	}
-
+
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/XMLAssert.java]: unnecessary change
